pkg/controller/pxc/tls.go

```diff
@@ -311,3 +318,57 @@ func (r *ReconcilePerconaXtraDBCluster) createSSLManualy(cr *api.PerconaXtraDBCl
 	}
 	return nil
 }
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileTLSToggle(ctx context.Context, cr *api.PerconaXtraDBCluster) error {
+	if cr.CompareVersionWith("1.16.0") < 0 {
+		return nil
+	}
+
+	annotationTLSState, ok := cr.Annotations[naming.AnnotationTLS]
```

Comment:
I would do this with a status field or with `status.conditions` rather than with an annotation